isolate/initialdispatch.go

```diff
@@ -268,6 +314,66 @@ func (d *initialDispatch) onSpawn(opts *cocaineProfile, name, executable string,
 	return newSpawnDispatch(d.ctx, cancelSpawn, prCh, &flagKilled, d.stream), nil
 }
 
+func (d *initialDispatch) onWorkersMetrics(uuidsQuery []string) (Dispatcher, error) {
+
+	log.G(d.ctx).Debugf("onWorkersMetrics() Uuids query (len %d): %s", len(uuidsQuery), strings.Join(uuidsQuery, ", "))
+
+	startTime := time.Now()
+
+	sendMetricsFunc := func(metrics MetricsResponse) {
+		var (
+			buf bytes.Buffer
+			err error
+		)
+
+		if d == nil {
+			log.G(d.ctx).Error("strange: dispatch is `nil`")
```

Comment:
can it happen? If no - maybe remove, so we can get a crash when logic is being somehow violated?